ladybug_geometry/geometry3d/_stl.py

```diff
@@ -0,0 +1,118 @@
+"""Support functions to turn stl into a Ladybug Mesh3D object.
+
+The methods in the object below are inspired from pySTL module here:
+https://github.com/proverbialsunrise/pySTL/blob/6afb180e14ee13fe313028cc9d9efa9104f7b1fe/pySTL.py
+"""
+
+import os
+import struct
+import warnings
+from .pointvector import Point3D
+
+try:
+    from itertools import izip as zip  # python 2
+except ImportError:
+    xrange = range  # python 3
+
+
+class STL:
+    """A class to represent an STL object.
+
+    Args:
+        file_path: Path to the STL file.
+    """
+
+    def __init__(self, file_path):
+        self._file_path = file_path
+        self._face_vertices = []
+        self._load_stl()
+
+    @property
+    def face_vertices(self):
+        """Get a list of faces where each face is a list of vertices for the face."""
+        return self._face_vertices
+
+    def _load_stl(self):
+        """Load the ASCII STL file and extract triangular faces."""
+
+        assert os.path.isfile(self._file_path), 'Failed to find %s' % self._file_path
+
+        with open(self._file_path, 'rb') as fp:
+            h = fp.read(80)
+            first_word = h[0:5]
+            fp.close()
```

Comment:
When you use `with` clause you don't need to close the file anymore. It does it for you.